include/common_robotics_utilities/print.hpp

```diff
@@ -27,9 +61,15 @@ inline std::string Print(const T& toprint,
 {
   CRU_UNUSED(add_delimiters);
   CRU_UNUSED(separator);
-  std::ostringstream strm;
-  strm << toprint;
-  return strm.str();
+#if COMMON_ROBOTICS_UTILITIES__SUPPORTED_ROS_VERSION == 2
```

Comment:
Since `Print` support is the only part that requires Rolling, what if we add a second `#define` to add this? Something like
```
// Print() support for ROS2 message types requires ROS2 Rolling. If using an older ROS2 distribution, use this define to disable.
// TODO(calderpg, hidmic) Remove this define once the necessary support lands in a stable ROS2 distribution.
#define COMMON_ROBOTICS_UTILITIES__SUPPORT_ROS2_MESSAGE_PRINT true
```
and then use both `COMMON_ROBOTICS_UTILITIES__SUPPORTED_ROS_VERSION` and `COMMON_ROBOTICS_UTILITIES__SUPPORT_ROS2_MESSAGE_PRINT` to enable use of `ROSMessagePrinter`?

Reply:
That sounds reasonable.

Reply:
Hmm, on a second thought, before Rolling there won't be support for printing ROS messages at all, so we may as well not do anything and let the compiler generate an error. That unless you want to keep the workaround I put together here for pre-Rolling distros.

Reply:
See https://github.com/calderpg/common_robotics_utilities/pull/14/commits/37d8c302e49f6adcbb99609e9d2ef49998e2ae94.

Reply:
You're right, having the compiler generate the error is good enough on pre-Rolling versions.